### ISSUE TITLE
matterbridge: add a download of configuration to the setup settings

### ISSUE THREAD

**Issue by jospoortvliet** [nextcloud/spreed#4100]
To make it easier for users to run matterbridge themselves, we can put a button or link on the bottom of the room configuration to download the configuration file. There should also be instructions on how to then run your own matterbridge with that on a desktop or server at home.

One issue is how to deal with the app password token. It is probably best to let the user generate a new app password manually and handle that themselves instead of trying to not delete it when they download the config?!?

If needed I can help with the docs and wording, ping me on Talk pls ;-)

**Comment by nickvergessen** [comment]
Well, so either you use the integration into Talk, then you don't need it, or you don't, then use the docs from https://github.com/42wim/matterbridge/

**Comment by eneiluj** [comment]
@nickvergessen I agree with that. It would still be helpful to provide the config to people having difficulties to write one on their own. This is surely out of the main scope but maybe nice for some users.

**Comment by jospoortvliet** [comment]
Yeah, it isn't a high prio issue, but it makes setting up a matterbridge yourself nicer and is probably not super hard to provide...

**Comment by nickvergessen** [comment]
But then you need to enable our matterbridge integration, configure all the rooms, download the config for each room and disable the integration. Doesn't sound very useful to me.

**Comment by jospoortvliet** [comment]
> But then you need to enable our matterbridge integration, configure all the rooms, download the config for each room and disable the integration. Doesn't sound very useful to me.

it's more for a case where a user wants to set some special configuration but there's no UI for that. Then he/she can take the pre-generated, add the flag, and run the binary on their desktop at home, or their own server. And 'user' here can be admin, of course.

Turning all matterbridges into manually run ones is not something I see a use case for. This is essentially an inbetween "us supporting every possible option" vs "do it yourself from scratch".

**Comment by nickvergessen** [comment]
Well but if you do it manually your own messages will not be mirrored. Unless you would use the credentials for the bot. But we can't give you those, as you could otherwise read all messages in all conversations that are mirrored.

**Comment by jospoortvliet** [comment]
> Well but if you do it manually your own messages will not be mirrored. Unless you would use the credentials for the bot. But we can't give you those, as you could otherwise read all messages in all conversations that are mirrored.

oh, that's another complication I didn't know about... So manually using matterbridge isn't as nice as using it through our integration?
